minerva_analysis/client/src/js/main.js

```diff
@@ -123,59 +136,60 @@ const actionChannelsToRenderChange = (d) => {
 eventHandler.bind(ChannelList.events.CHANNELS_CHANGE, actionChannelsToRenderChange);
 
 /**
- * Listen to regional or single cell selection
- * @param  {package object} d The selections
+ * Listen to regional or single cell selection.
+ *
+ * @param d - The selections
  */
 const actionImageClickedMultiSel = (d) => {
     d3.select('body').style('cursor', 'progress');
     // add newly clicked item to selection
-    if (!Array.isArray(d.selectedItem)) {
-        dataLayer.addToCurrentSelection(d.selectedItem, true, d.clearPriors);
+    if (!Array.isArray(d.item)) {
+        dataLayer.addToCurrentSelection(d.item, true, d.clearPriors);
+        updateSeaDragonSelection([d.item]);
     } else {
-        dataLayer.addAllToCurrentSelection(d.selectedItem);
+        dataLayer.addAllToCurrentSelection(d.item);
+        updateSeaDragonSelection(d.item);
     }
-    updateSeaDragonSelection();
     d3.select('body').style('cursor', 'default');
 }
 eventHandler.bind(ImageViewer.events.imageClickedMultiSel, actionImageClickedMultiSel);
 
 /**
- * Listen to Channel Select Click Events
- * @param  {package object} d The selected/deselected channels
+ * Listen to Channel Select Click Events.
+ *
+ * @param sels - The selected/deselected channels
  */
 const channelSelect = async (sels) => {
-    // pause new rendering until data loads
-    const resume = seaDragonViewer.sleep(); 
+    updateSeaDragonSelection();
     let channelCells = await dataLayer.getChannelCellIds(sels);
     dataLayer.addAllToCurrentSelection(channelCells);
-    updateSeaDragonSelection();
-    resume();
 }
 eventHandler.bind(ChannelList.events.CHANNEL_SELECT, channelSelect);
 
 /**
- * Listens to and updates based on selection changes (specific for seadragon)
- * @param  {boolean} d Whether to repaint
+ * Listens to and updates based on selection changes (specific for seadragon).
+ *
+ * @param ids - list of selected ids
  */
-function updateSeaDragonSelection() {
-    seaDragonViewer.updateSelection(dataLayer.getCurrentSelection());
+function updateSeaDragonSelection(ids=[]) {
+    const { idField } = this.config[datasource].featureData[0];
+    seaDragonViewer.pickedIds = ids.map(v => v[idField]);
+    seaDragonViewer.forceRepaint();
 }
 
 /**
- * Listens to and updates based on selection changes (specific for seadragon)
- * @param  {boolean} d Whether to repaint
+ * Listens to and updates based on selection changes (specific for seadragon).
+ *
+ * @param packet - gating filter
  */
 const gatingBrushEnd = async (packet) => {
-    // Init gated cells
-    let gatedCells = [];
-    // Get custom cell ids (made-to-order properties)
+    return updateSeaDragonSelection(); // TODO, compatibility
+    /*
```

Comment:
I've commented-out the server-side gating calls, as they're not used on this branch. For compatibility with other branches, I can explore how to leverage both client and server-side together.